AppCenterDataStorage/AppCenterDataStorage/MSDataStore.m

```diff
@@ -668,4 +673,115 @@ - (void)authTokenContext:(MSAuthTokenContext *)__unused authTokenContext didUpda
+                                                     writeOptions:(MSWriteOptions * _Nullable) writeOptions
+                                                completionHandler:^(MSDocumentWrapper *documentWrapper) {
+                                                  if (!documentWrapper.error) {
+                                                    if (writeOptions.deviceTimeToLive > kMSDataStoreTimeToLiveNoCache ||
+                                                        writeOptions.deviceTimeToLive == kMSDataStoreTimeToLiveInfinite) {
+                                                      [self.dataOperationProxy.documentStore upsertWithToken:tokenResponses.tokens[0]
+                                                                                             documentWrapper:documentWrapper
+                                                                                                   operation:nil
+                                                                                            deviceTimeToLive:writeOptions.deviceTimeToLive];
+
+                                                    } else {
+                                                      [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
+                                                                                                  documentId:operation.documentId];
+                                                    }
+
+                                                  } else if (documentWrapper.error.errorCode == MSACDocumentNotFoundErrorCode ||
+                                                             documentWrapper.error.errorCode == MSACDocumentConflictErrorCode) {
+                                                    MSLogError(
+                                                        [MSDataStore logTag],
+                                                        @"Failed to call Cosmos delete API. Remote operation failed with error code: %ld",
+                                                        documentWrapper.error.errorCode);
+                                                    [self.dataOperationProxy.documentStore deleteWithToken:tokenResponses.tokens[0]
+                                                                                                documentId:operation.documentId];
+                                                  } else {
+                                                    MSLogError([MSDataStore logTag], @"Failed to call Cosmos create or replace API: %@",
+                                                               [documentWrapper.error.error localizedDescription]);
+                                                  }
+                                                }];
+                                       } else if ([operation.operation isEqualToString:kMSPendingOperationDelete]) {
+
+                                         // Perform delete operation
```

Comment:
My understanding is we need to remove the pending operation from local cache the operation has expired. I don't see any logic in codepath the DELETE operation. Did I miss anything?

Reply:
@dhei  we have if (not expatriated check) else (delete from cache via deleteWithToken method) in both places in the method that you review.